From 8b01307f77ed2ab7116853d25a4bc3b93c63e7c6 Mon Sep 17 00:00:00 2001
From: Leonard Crestez <leonard.crestez@nxp.com>
Date: Thu, 7 Jun 2018 15:26:09 +0300
Subject: [PATCH 3906/5242] MLK-18537 ARM: dts: imx6qdl-sabreauto: Move
 sensors under i2c mux

commit  b645137625feeaa9cf9546379ab02e3bffecb650 from
https://source.codeaurora.org/external/imx/linux-imx.git

Suspend fails on sabreauto because the i2c mux is set to the default
state before suspending sensors:

dpm_run_callback(): isl29023_suspend+0x0/0x3c returns -11
PM: Device 2-0044 failed to suspend: error -11
PM: Some devices failed to suspend, or early wake event detected

Fix this by moving sensons to i2cmux/i2c@1 just like the rest of I2C3
devices. This is a porting issue, in imx_4.9.y there is no dynamic
muxing for i2c3.

Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Acked-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6qdl-sabreauto.dtsi |   50 +++++++++++++++---------------
 1 file changed, 25 insertions(+), 25 deletions(-)

diff --git a/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi b/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
index d670a6d..6e03d75 100644
--- a/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
@@ -347,6 +347,31 @@
 				gpio-controller;
 				#gpio-cells = <2>;
 			};
+
+			isl29023@44 {
+				compatible = "fsl,isl29023";
+				reg = <0x44>;
+				rext = <499>;
+				interrupt-parent = <&gpio5>;
+				interrupts = <17 2>;
+			};
+
+			mag3110@0e {
+				compatible = "fsl,mag3110";
+				reg = <0x0e>;
+				position = <2>;
+				interrupt-parent = <&gpio2>;
+				interrupts = <29 1>;
+			};
+
+			mma8451@1c {
+				compatible = "fsl,mma8451";
+				reg = <0x1c>;
+				position = <7>;
+				interrupt-parent = <&gpio6>;
+				interrupts = <31 8>;
+				interrupt-route = <1>;
+			};
 		};
 	};
 
@@ -641,31 +666,6 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_i2c3>;
 	status = "okay";
-
-	isl29023@44 {
-		compatible = "fsl,isl29023";
-		reg = <0x44>;
-		rext = <499>;
-		interrupt-parent = <&gpio5>;
-		interrupts = <17 2>;
-	};
-
-	mag3110@0e {
-		compatible = "fsl,mag3110";
-		reg = <0x0e>;
-		position = <2>;
-		interrupt-parent = <&gpio2>;
-		interrupts = <29 1>;
-	};
-
-	mma8451@1c {
-		compatible = "fsl,mma8451";
-		reg = <0x1c>;
-		position = <7>;
-		interrupt-parent = <&gpio6>;
-		interrupts = <31 8>;
-		interrupt-route = <1>;
-	};
 };
 
 &iomuxc {
-- 
1.7.9.5

